src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -31,184 +31,387 @@ public static class SchemaConverter
-            }
-
             // Top-level object type definition name should be singular.
             // The singularPlural.Singular value is used, and if not configured,
             // the top-level entity name value is used. No singularization occurs
             // if the top-level entity name is already plural.
             return new ObjectTypeDefinitionNode(
                 location: null,
-                name: nameNode,
+                name: new(value: GetDefinedSingularName(entityName, configEntity)),
                 description: null,
-                objectTypeDirectives,
+                directives: GenerateObjectTypeDirectivesForEntity(entityName, configEntity, rolesAllowedForEntity),
                 new List<NamedTypeNode>(),
                 fields.Values.ToImmutableList());
         }
 
+        /// <summary>
+        /// Helper method to generate the FieldDefinitionNode for a column in a table/view or a result set field in a stored-procedure.
+        /// </summary>
+        /// <param name="configEntity">Entity's definition (to which the column belongs).</param>
+        /// <param name="columnName">Backing column name.</param>
+        /// <param name="column">Column definition.</param>
+        /// <param name="directives">List of directives to be added to the column's field definition.</param>
+        /// <param name="roles">List of roles having read permission on the column (for tables/views) or execute permission for stored-procedure.</param>
+        /// <returns>Generated field definition node for the column to be used in the entity's object type definition.</returns>
+        private static FieldDefinitionNode GenerateFieldForColumn(Entity configEntity, string columnName, ColumnDefinition column, List<DirectiveNode> directives, IEnumerable<string>? roles)
+        {
+            if (GraphQLUtils.CreateAuthorizationDirectiveIfNecessary(
+                                            roles,
+                                            out DirectiveNode? authZDirective))
+            {
+                directives.Add(authZDirective!);
+            }
+
+            string exposedColumnName = columnName;
+            if (configEntity.Mappings is not null && configEntity.Mappings.TryGetValue(key: columnName, out string? columnAlias))
+            {
+                exposedColumnName = columnAlias;
+            }
+
+            NamedTypeNode fieldType = new(GetGraphQLTypeFromSystemType(column.SystemType));
+            FieldDefinitionNode field = new(
+                location: null,
+                new(exposedColumnName),
+                description: null,
+                new List<InputValueDefinitionNode>(),
+                column.IsNullable ? fieldType : new NonNullTypeNode(fieldType),
+                directives);
+            return field;
+        }
+
+        /// <summary>
+        /// Helper method to generate field for a relationship for an entity. While processing the relationship, it does some other things:
```

Comment:
can you elaborate more here on why this helper is signification? a relationship field has (relationship, foreign key) directive to document cardinality,

Reply:
Expanded comment.